packages/ai/src/evals/eval.ts

```diff
@@ -497,26 +498,69 @@ async function registerEval<
                       },
                     },
                     async (scorerSpan) => {
-                      const start = performance.now();
-                      const result = await scorer({
-                        input: data.input,
-                        output: output,
-                        expected: data.expected,
-                      });
-
-                      const duration = Math.round(performance.now() - start);
-                      const scoreValue = result.score as number;
-
-                      scorerSpan.setAttributes({
-                        [Attr.Eval.Score.Name]: scorerName,
-                        [Attr.Eval.Score.Value]: scoreValue,
-                      });
-
-                      return {
-                        name: scorerName,
-                        ...result,
-                        metadata: { duration, startedAt: start, error: null },
-                      };
+                      try {
+                        const start = performance.now();
+                        const result = await scorer({
+                          input: data.input,
+                          output: output,
+                          expected: data.expected,
+                        });
+
+                        const duration = Math.round(performance.now() - start);
+                        const scoreValue = result.score as number;
+                        const metadata = Object.assign(
+                          { duration, startedAt: start },
+                          result.metadata,
+                        );
+
+                        scorerSpan.setAttributes({
+                          [Attr.Eval.Score.Name]: scorerName,
+                          [Attr.Eval.Score.Value]: scoreValue,
+                          [Attr.Eval.Score.Metadata]: JSON.stringify(metadata),
+                        });
+
+                        if (metadata.error) {
+                          const msg = getErrorMessage(metadata.error);
+
+                          scorerSpan.setStatus({
+                            code: SpanStatusCode.ERROR,
+                            message: msg,
+                          });
+                        }
+
+                        return {
+                          name: scorerName,
+                          score: scoreValue,
+                          metadata: Object.assign({ duration, startedAt: start }, result.metadata),
+                        };
+                      } catch (error) {
+                        console.error(`ERROR: scorer ${scorerName} failed. Cause: \n`, error);
+                        const msg = getErrorMessage(error);
+                        const metadata = {
+                          duration,
```

Comment:
This duration is not coming from the right place, need to define the start outside of the trycatch